packages/app/src/components/swap/swap.tsx

```diff
@@ -92,10 +95,17 @@ const Swap: FC<Props> = ({
         }
     ) ?? [];
 
-  const handleSendTokenInput = (e: any) => {
+  const handleSendTokenInput: ChangeEventHandler<HTMLInputElement> = useCallback((e) => {
     setSendTokenAmount(e.target.value);
-    setReceiveTokenAmount(calcOutGivenIn);
-  };
+  }, []);
+
+  useEffect(() => {
+    if (calcOutGivenIn != null && !calcOutGivenIn.eq(receiveTokenAmount)) {
```

Comment:
`calcOutGivenIn` can never be `null`

even it it were you could use `calcOutGivenIn?.eq(...)` as a shorthand.